### DISCUSSION TITLE
questions about the distance sensor

### DISCUSSION THREAD

**Discussion by wjh-scut** [microsoft/AirSim#4249] [General]
I set up a distance sensor for the UAV to measure the distance between the UAV and the cylindrical obstacle, as shown below. But I found a problem: why did the green dot of the distance sensor not fall on the surface of the cylinder, but on a certain position in front of the cylinder? Is there something wrong with my cylinder? Or something else?

![微信截图_20211229111813](https://user-images.githubusercontent.com/28137201/147624504-0cc5d461-bf5a-42fc-986c-fd6a08f76dbe.png)

Any help?

settings:
```
{
  "SeeDocsAt": "https://github.com/Microsoft/AirSim/blob/master/docs/settings.md",
  "SettingsVersion": 1.2,
  "SimMode": "Multirotor",
  "Vehicles": {
    "UAV": {
      "VehicleType": "SimpleFlight",
      "X": 0,
      "Y": 0,
      "Z": 0,
      "Yaw": 0,
      "Sensors": {
        "Distance1": {
          "SensorType": 5,
          "Enabled": true,
          "DrawDebugPoints": true
        }
      }

    }
  }
}
```

**Comment by rajat2004** [comment]
Might be that the distance is greater than 40m which is the default max distance measured by [distance sensor](https://microsoft.github.io/AirSim/distance_sensor/). Try getting the sensor data to verify this. You can set the `MaxDistance` field in the sensor params to change this

**Reply by wjh-scut** [reply]
I got the sensor data and confirmed that the distance was not beyond the maximum distance. Is something else wrong?

**Reply by ghost** [reply]
How did you get the sensor data?
